floodmodeller_api/toolbox/model_build/structure_log/structure_log.py

```diff
@@ -1,14 +1,46 @@
+import copy
 import csv
 
 from floodmodeller_api import DAT
 
 
+def extract_attrs(object_input, keys):
+    """
+    extract attributes and their values from a given object and return as a dict
+
+    No handling for where attributes dont exist but can ignore for when this is an issue, shouldnt be an issue here
+    """
+    return {key: getattr(object_input, key) for key in keys}
+
+
+def serialise_keys(old_dict):
+    """
+    dictionary keys must be unique; but in FM the node label isnt always unique
+    however the (label,type) pair will always be unique; so we have to use that as our key.
+    json doesnt like tuples as keys so we have to convert them first if we need to serialise our dictionary output.
+
+    Replaces tuple keys with string versions of themselves, wrapped in ()
+
+    """
+    new_dict = {}
```

Comment:
Can you double check that you can't write `return {f"({','.join(key)})": value for key, value in old_dict.items()}` here instead (making sure to use that combination of single and double quotes)? I think this would fail a new linting check we just added to the main branch yesterday

Reply:
ammended, I think this is slightly different than what I originally had as this one doesnt break the linting.